src/looprestoration.rs

```diff
@@ -372,64 +376,49 @@ unsafe fn wiener_rust<BD: BitDepth>(
         edges,
     );
 
-    let mut hor: [uint16_t; 27300] = [0; 27300];
-    let mut hor_ptr: *mut uint16_t = hor.as_mut_ptr();
-    let filter: *const [int16_t; 8] = ((*params).filter.0).as_ptr();
+    // Values stored between horizontal and vertical filtering don't
+    // fit in a uint8_t.
+    let mut hor = [0u16; 70 /*(64 + 3 + 3)*/ * REST_UNIT_STRIDE];
+
+    let filter = &params.filter.0;
     let bitdepth = bd.bitdepth().as_::<libc::c_int>();
-    let round_bits_h = 3 as libc::c_int + (bitdepth == 12) as libc::c_int * 2;
-    let rounding_off_h = (1 as libc::c_int) << round_bits_h - 1;
-    let clip_limit = (1 as libc::c_int) << bitdepth + 1 + 7 - round_bits_h;
-    let mut j = 0;
-    while j < h + 6 {
-        let mut i = 0;
-        while i < w {
-            let mut sum = (1 as libc::c_int) << bitdepth + 6;
+    let round_bits_h = 3 + (bitdepth == 12) as libc::c_int * 2;
+    let rounding_off_h = 1 << round_bits_h - 1;
+    let clip_limit = 1 << bitdepth + 1 + 7 - round_bits_h;
+    let chunks = tmp
+        .chunks_exact(REST_UNIT_STRIDE)
```

Comment:
I added a `.ext_as_chunks()` method, which is the unstable std method added in our code to make it stable.  It's an array version of chunks basically, so there won't be bounds checking on the array chunks.

Reply:
Generally I'd recommend splitting these kinds of changes.  There's the basic cleanup ones that are unlikely to change any semantics or cause errors, then there's cast elision which in certain cases could lead to overflow errors, there's conversion to safe references which is straightforward usually, there's conversion to safe slices, which is more complex when we have to guess the size, and then there's these bounds checking optimizations.  I think it's better to split them into separate commits since there's very different chances of errors for each.  IMO it's also easier to follow the changes that way.

Reply:
Is `ext_as_chunks` in the project? I'm not seeing it. Or do you mean that you plan to add it?

Reply:
Oh, I was in the middle of a PR adding it.  I think I didn't get to opening it yet.

Reply:
Okay then let's hold off on updating this PR to use `as_chunks`. We can come back and clean this up after you've had a chance to add a stable version of `as_chunks`, but I don't don't want to use unstable features in the meantime.